provider/internal/java/dependency.go

```diff
@@ -138,9 +141,20 @@ func (p *javaServiceClient) GetDependenciesFallback(ctx context.Context, locatio
 		if *d.Version != "" {
 			if strings.Contains(*d.Version, "$") {
 				version := strings.TrimSuffix(strings.TrimPrefix(*d.Version, "${"), "}")
-				version = pom.Properties.Entries[version]
-				if version != "" {
+				p.log.V(10).Info("Searching for property in properties",
+					"property", version,
+					"properties", pom.Properties)
+				if pom.Properties == nil {
+					p.log.Info("Cannot resolve version property value as POM does not have properties",
```

Comment:
very very small nit: what do we think of making the this `..."groupID", pomCo..., "artifactID", pomCor....)` 

Just wondering what folks think

Reply:
The pomCoordinate function was just a quick hack to cope with strings which are either set or "". Any better approach is highly appreciated.

Reply:
Oh, I was not suggesting changing the helper function, I think that for this particular thing, it is a nice UX boost. I was discussing of making the log statement use the key, value pairs for groupID and artifactID instead of combining in a single key(POM). Just think it might be cleaner.